src/data/equiv/embedding.lean

```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2021 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import logic.embedding
+import data.set.lattice
+
+/-!
+# Equivalences on embeddings
+
+This file shows some advanced equivalences on embeddings, useful for constructing larger
+embeddings from smaller ones.
+-/
+
+
+open function.embedding
+
+namespace equiv
+
+/-- Embeddings from a sumtype are equivalent to two separate embeddings with disjoint ranges. -/
+def sum_embedding_equiv_prod_embedding_disjoint {α β γ : Type*} :
+((α ⊕ β) ↪ γ) ≃ {f : (α ↪ γ) × (β ↪ γ) // disjoint (set.range f.1) (set.range f.2)} :=
+{ to_fun := λ f, ⟨(inl.trans f, inr.trans f),
+  begin
+    rintros _ ⟨⟨a, h⟩, ⟨b, rfl⟩⟩,
+    simp only [trans_apply, inl_apply, inr_apply] at h,
+    have : sum.inl a = sum.inr b := f.injective h,
+    simp only at this,
+    assumption
+  end⟩,
+
```

Comment:
We don't usually leave blank lines between fields.

Reply:
Still needs fixing.

Reply:
Oops, yep, on it